solutions/20.js

```diff
@@ -0,0 +1,12 @@
+const solution = (current, next = [], result = []) => {
+  if (current.length == 0 && next.length == 0) return result;
+  if (current.length == 0) {
+    result.push(next);
+    const n = current.shift();
+    current = next;
+    next = [];
+  }
+  return solution(current, next.concat(n.children), result);
```

Comment:
Do we need to do `n = current.shift()` outside the if statement?

Reply:
I just took off the else statement. With this change, it technically already is "outside" of the if statement! 👍